File: src/main/scala/com/github/mauricio/postgresql/QueryResult.scala
package com.github.mauricio.postgresql

/**
 * User: Maurício Linhares
 * Date: 3/3/12
 * Time: 4:01 PM
 */

class QueryResult ( val rowsAffected : Int, val statusMessage : String, val rows : Option[Query] ) {

  override def toString : String = {
    "QueryResult{rows -> %s,status -> %s}".format( this.rowsAffected, this.statusMessage )
  }

}

File: db-async-common/src/main/java/com/github/jasync/sql/db/QueryResult.kt
package com.github.jasync.sql.db

/**
 *
 * This is the result of the execution of a statement, contains basic information as the number or rows
 * affected by the statement and the rows returned if there were any.
 *
 * @param rowsAffected
 * @param statusMessage
 * @param rows
 */

open class QueryResult(val rowsAffected: Long, val statusMessage: String?, val rows: ResultSet = EMPTY_RESULT_SET) {

  override fun toString() : String {
    return "QueryResult{rows -> %s,status -> %s}".format(this.rowsAffected, this.statusMessage)
  }

}
